patroni/ha.py

```diff
@@ -757,10 +763,8 @@ def _run_cycle(self):
                 if self.cluster.is_unlocked():
                     return self.process_unhealthy_cluster()
                 else:
-                    msg = self.evaluate_scheduled_restart()
-                    if msg is not None:
-                        return msg
-                    return self.process_healthy_cluster()
+                    msg = self.process_healthy_cluster()
+                    return not self._async_executor.busy and self.evaluate_scheduled_restart() or msg
```

Comment:
I find this kind of `condition and side_effect() or alternate_result` idiom somewhat hard to make sense of when reading code. I think moving the `_async_executor.busy` check into `evaluate_scheduled_restart()` would clean this up further:

```python
msg = self.process_healthy_cluster()
return self.evaluate_scheduled_restart() or msg
```